src/lua/init.lua

```diff
@@ -169,22 +198,19 @@ local function rocks_loader_func(lib)
     end
 end
 
-local function search_path_func(cpath)
-    return function(name)
-        return package.searchpath(name, cpath and package.cpath or package.path)
-    end
-end
+local search_func_defs = {
```

Comment:
`package.search()` did look into search_cwd_lib, search_rocks_lib and so on. Now it does not. Why?

Reply:
Didn't get you. There is no more `search_cwd_lib` and `search_rocks_lib` functions anymore. But the behaviour is actually the same (besides new "search root" functionality added)

Reply:
If you are saying about `search_cwd_lua` and `search_rocks_lua` then it was removed on purpose. See, `package.search` is used in Tarantool only for loading C functions (dlopen). I don't know why this function is in public interface or even why it was written in Lua space.

With that said when loading a C function, if coincidentally `package.search` finds a lua module, than following dlopen will fail. I made a simple test case when it may happen:

```
- error: "Failed to dynamically load module '': dlopen(/tmp/tntX4ScRS/function1.dylib,
    6): no suitable image found.  Did find:\n\t/tmp/tntX4ScRS/function1.dylib: file
    too short\n\t/Users/i.konyuhov/Workspace/tarantool/test/box/function1.lua: file
    too short"
```

In my opinion, `package.search` should not even look at the lua paths, but on the other hand, since `package.search` is a public function, users may have using it somehow (it is not in docs though). I tend to the first option, what do you thing?

Reply:
There is https://github.com/tarantool/avro-schema/commit/a8da39ab44fd007989f126592792655b50db2d69 (which however was reverted in https://github.com/tarantool/avro-schema/commit/e8faf0c1fbe0f700b21c02d15f3dbd37b2bec4a4). Here it is used for loading C library, so it is ok. Also I see that this function was mentioned twice in tarantool's public telegram chat.

I would not change the behaviour for now even despite that this looks buggy in some cases.

Reply:
Returned code, which is looking for lua dependencies as well as C dependencies

Reply:
Returned back to previous behaviour